### ISSUE TITLE
Inference hardware requirements

### ISSUE THREAD

**Issue by abrichr** [OSU-NLP-Group/UGround#2]
Hello, and thank you for the excellent work!

In the paper it says:

> The first stage takes about 50 hours on a single 4x NVIDIA A100 machine (global batch size 128 with gradient
accumulation). And for the large scale GUI data training, we use 112 NVIDIA H100 GPUs and finish the
training in about 6 hours (global batch size 448).

Can you please clarify what are the inference time hardware requirements? Any chance of running this on CPU?

Thanks again!

**Comment by boyugou** [comment]
Overall, it's built on LLava with slight adaptations (mainly about input image processing), so it's definitely possible to run on a CPU (take [Ollama](https://ollama.com/library/llava) as a reference).

**Comment by dprokhorov17** [comment]
@abrichr I'm running this within a docker container and my memory footprint is the following:

<img width="505" alt="image" src="https://github.com/user-attachments/assets/a169154c-c832-4782-8ebd-310984a2f7f7">

It's running on `bf16` float precision using `cached=True`

**Comment by GuoHaoren** [comment]
Hi, I am implement the inference "single_infer.py" on my mac with CPU only. It is very slow. Is there any suggestion on the configurations settings regarding to "load_pretrained_model", "tokenizer_image_token" and "model.generate()".

**Comment by boyugou** [comment]
@GuoHaoren  @abrichr @dprokhorov17 

We have trained a stronger while smaller 2B model:

[osunlp/UGround-V1-2B](https://huggingface.co/osunlp/UGround-V1-2B)

I'm still trying to see what's the best way to run on CPU. 

Maybe by quantization like 
GGUF: https://huggingface.co/mradermacher/UGround-V1-2B-GGUF
or AWQ/GPQ which are suggested by Qwen2-VL

**Comment by dprokhorov17** [comment]
@boyugou and still, you didn't provide any finetuning/training script in order to be able to reproduce your results.

**Comment by boyugou** [comment]
> @boyugou and still, you didn't provide any finetuning/training script and most important the data in order to be able to reproduce your results.

Let me clarify a little bit. Here are the main training script and code for the previous training:

https://github.com/boyugou/llava_uground/blob/90ff02d24c3f8c7a9fb5c90050fa003b0512910f/scripts/ui_v1/finetune_task_lora.sh
https://github.com/boyugou/llava_uground/blob/90ff02d24c3f8c7a9fb5c90050fa003b0512910f/llava/train/train_s3.py

The Qwen2-VL-based models are trained by the infra of [Mosaic ML](https://github.com/mosaicml/llm-foundry), using a in-house codebase which I cannot share with you. But I will share the yaml file for the details of hyper-parameters used. 

The data used for the above two are totally the same

**Comment by boyugou** [comment]
> @boyugou and still, you didn't provide any finetuning/training script and most important the data in order to be able to reproduce your results.

Regarding the data, plz give me a bit more time. And I will release the code I used first. For big companies, it should be easy to collect better data than what I used (Web-Hybrid) with the same pipeline, by using:
- better webpage url list (I randomly sampled from Common Crawl)
- better captioning MLLM and rewriting LM

**Comment by boyugou** [comment]
Do these answer your question? @dprokhorov17 Hope to get them ready for everyone soon.

**Comment by boyugou** [comment]
> [@boyugou](https://github.com/boyugou) and still, you didn't provide any finetuning/training script and most important the data in order to be able to reproduce your results.

Hi all,

We have released the data for research use: https://huggingface.co/datasets/osunlp/UGround-V1-Data

Feel free to have a try.

Here's my tweet for the release: https://x.com/BoyuGouNLP/status/1882609270731972895

The code will be ready soon as well 😊

Sorry for the delay and thanks for your patience.
